utils/http.go

```diff
@@ -28,74 +29,137 @@ type rootHandler struct {
 	context context.Context
 	trust   signed.CryptoService
 	//cachePool redis.Pool
+	imageNameAt ImageNameLocation
+}
+
+// ImageNameLocation is used to define where the auth code should read the image name
+// from. For most repo operations, it will be ImageInURL, i.e. it's the segments between
+// `/v2/` and `/_trust`. For the changefeed however, an image name is optional, and if
```

Comment:
i.e., it's the segments between "/v2/" and "/_trust" - `/v2/docker.io/official/busybox/_trust/_tuf/targets.json`.